frontend/src/components/grantaccess/GrantAccess.vue

```diff
@@ -0,0 +1,402 @@
+<script setup lang="ts">
+import { onMounted, ref } from 'vue';
+import router from '@/router';
+import { Form as VeeForm, Field, ErrorMessage } from 'vee-validate';
+import { number, object, string } from 'yup';
+
+import Button from 'primevue/button';
+import InputText from 'primevue/inputtext';
+import Dropdown from 'primevue/dropdown';
+import RadioButton from 'primevue/radiobutton';
+import ForestClientCard from './ForestClientCard.vue';
+
+import { ApiServiceFactory } from '@/services/ApiServiceFactory';
+import {
+    selectedApplication,
+    selectedApplicationDisplayText,
+} from '@/store/ApplicationState';
+import {
+    grantAccessFormData,
+    domainOptions,
+    FOREST_CLIENT_INPUT_MAX_LENGTH,
+    setGrantAccessFormData,
+    getGrantAccessFormData,
+} from '@/store/GrantAccessDataState';
+
+import type { FamApplicationRole, FamForestClient } from 'fam-api';
+
+interface IFormData {
```

Comment:
why we need a new interface type? is it different than this type "import type { FamUserRoleAssignmentCreate } from 'fam-api'" in the GrantAccessDataState.ts file?